packages/router-plugin/src/core/router-code-splitter-plugin.ts

```diff
@@ -66,6 +66,10 @@ export const unpluginRouterCodeSplitterFactory: UnpluginFactory<
   let ROOT: string = process.cwd()
   let userConfig: Config
 
+  // Cache of shared module-level variables that were exported in reference files
+  // Key: base filename (without query params), Value: Set of exported identifier names
+  const sharedExportsCache = new Map<string, Set<string>>()
```

Comment:
how does this work with HMR? 

also how does this work across multiple vite environments? i assume this will work since we have the plugin per environment?